contracts/TxRelay.sol

```diff
@@ -61,4 +69,44 @@ contract TxRelay {
     function getNonce(address add) public constant returns (uint) {
         return nonce[add];
     }
+
+    /*
+     * @dev Adds a number of addresses to a specific whitelist. Only
+     * the owner of a whitelist can add to it.
+     * @param listNum The list number
+     * @param allowedSenders the addresses to add to the whitelist
+     */
+    function addToWhitelist(uint listNum, address[] allowedSenders) public {
+        if (listOwner[listNum] == 0x0) {
+            listOwner[listNum] = msg.sender;
+        }
+        updateWhitelist(listNum, allowedSenders, true);
+    }
+
+    /*
+     * @dev Removes a number of addresses from a specific whitelist. Only
+     * the owner of a whitelist can remove from it.
+     * @param listNum The list number
+     * @param allowedSenders the addresses to add to the whitelist
+     */
+    function removeFromWhitelist(uint listNum, address[] allowedSenders) public {
+        updateWhitelist(listNum, allowedSenders, false);
+    }
+
+    /*
+     * @dev Internal logic to update a whitelist
+     * @param listNum The list number
+     * @param allowedSenders the addresses to add to the whitelist
+     * @param onList whether to add or remove addresses
```

Comment:
`onList` is a bit confusing as well. It seems to talk about the existence of these addresses on the whitelist already, rather than whether they are being added or removed. Maybe change to something like`senderNewStatus` or something similar.

Reply:
thanks, I was looking for a better name but couldn't come up with one.